keyboards/lily58/keymaps/defcxz/keymap.c

```diff
@@ -0,0 +1,243 @@
+ *                   |      |      |      |/       /         \      \ |      |      |      |
+ *                   `----------------------------'           '------''--------------------'
+ */
+ [_QWERTY] = LAYOUT(
+  KC_ESC,   KC_1,   KC_2,    KC_3,    KC_4,    KC_5,                     KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
+  KC_TAB,   KC_Q,   KC_W,    KC_E,    KC_R,    KC_T,                     KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC,
+  KC_LSFT,  KC_A,   KC_S,    KC_D,    KC_F,    KC_G,                     KC_H,    KC_J,    KC_K,    KC_L,    KC_UP,   KC_RSFT,
+  KC_LCTRL,  KC_Z,   KC_X,   KC_C,    KC_V,    KC_B, KC_QUOT,  KC_BSLS,  KC_N,    KC_M,    KC_SCLN, KC_LEFT, KC_DOWN, KC_RIGHT,
+                        MO(_LOWER),   KC_LGUI, TD(TD_LALT_CAPS), KC_SPC, KC_ENT,  KC_COMM, KC_DOT,  KC_MINS
+),
+//
+/* LOWER
+ * ,-----------------------------------------.                    ,-----------------------------------------.
+ * |      |      |      |      |      |      |                    |      |      |      |      |      |      |
+ * |------+------+------+------+------+------|                    |------+------+------+------+------+------|
+ * |  F1  |  F2  |  F3  |  F4  |  F5  |  F6  |                    |  F7  |  F8  |  F9  | F10  | F11  | F12  |
+ * |------+------+------+------+------+------|                    |------+------+------+------+------+------|
+ * |   `  |   !  |   @  |   #  |   $  |   %  |-------.    ,-------|   ^  |   &  |   *  |   (  |  Ñ   |   -  |
+ * |------+------+------+------+------+------|   [   |    |    ]  |------+------+------+------+------+------|
+ * |      |      |      |      |      |      |-------|    |-------|      |   _  |   +  |   {  |   }  |   |  |
+ * `-----------------------------------------/       /     \      \-----------------------------------------'
+ *                   | LAlt | LGUI |LOWER | /Space  /       \Enter \  |RAISE |BackSP| RGUI |
+ *                   |      |      |      |/       /         \      \ |      |      |      |
+ *                   `----------------------------'           '------''--------------------
+ */
+//!!!!"""""####"
+[_LOWER] = LAYOUT(
+  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,                     KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,
+  _______, KC_PGUP, KC_RBRC, _______, _______, _______,                   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
+  KC_LSFT, KC_PGDN, KC_PSCR, _______, RGB_SAI, RGB_SAD,                   KC_F13,  KC_F14,  KC_F15,  KC_F16,  KC_F17,  KC_SLSH,
+  RGB_TOG, RGB_MOD, RGB_VAI, RGB_VAD, RGB_HUI, RGB_HUD, _______, _______, KC_F19,  KC_F20,  KC_F21,  KC_F22,  KC_F23,  KC_F24,
+                             _______, KC_CAPS, _______, _______, _______, KC_DEL, KC_PLUS, KC_EQL
+),
+};
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
+
```

Comment:
This core [tri layer](https://docs.qmk.fm/#/feature_tri_layer) should be used here instead.
```suggestion
```

Reply:
Either may be fine, but as MO is just being used, the tri layer keycodes would probably be better, yeah